lib/invoice_item_repository.rb

```diff
@@ -121,4 +121,27 @@ def find_invoice(id)
   def find_item(item_id)
     sales_engine.find_item_by_id(item_id)
   end
+
+  def create_new_items(items, id)
+    items.each do |item|
+      grouped_items = items.group_by do |item|
+        item
+      end
+      quantity = grouped_items.map do |item|
```

Comment:
Shadowing outer local variable - `item`.<br>Pass `&:count` as an argument to `map` instead of a block.